freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/ChildEnvironmentService.java

```diff
@@ -69,6 +70,11 @@ public void detachChildEnvironment(DetachChildEnvironmentRequest request, String
                             request.getParentEnvironmentCrn(), request.getChildEnvironmentCrn())));
         MDCBuilder.buildMdcContext(childEnvironment.getStack());
         LOGGER.info("Detaching child env: {}", childEnvironment);
-        repository.delete(childEnvironment);
+        try {
+            repository.delete(childEnvironment);
+        } catch (ObjectOptimisticLockingFailureException e) {
+            LOGGER.info("Child env {} is already detached", childEnvironment);
```

Comment:
logging the actual exception would be nice

Reply:
Nice catch, I have opened a follow up PR: https://github.com/hortonworks/cloudbreak/pull/12215